server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -696,6 +702,41 @@ static String getCommitCheckpointFileName(long generation) {
         return TRANSLOG_FILE_PREFIX + generation + CHECKPOINT_SUFFIX;
     }
 
+    /**
+     * Trims translog for terms of files below <code>belowTerm</code> and seq# above <code>aboveSeqNo</code>.
+     * Effectively it moves max visible seq# {@link Checkpoint#trimmedAboveSeqNo} therefore {@link TranslogSnapshot} skips those operations.
+     */
+    public void trimOperations(long belowTerm, long aboveSeqNo) throws IOException {
+        assert aboveSeqNo >= 0 : "aboveSeqNo has to be non-negative number";
+
+        try (ReleasableLock lock = writeLock.acquire()) {
+            ensureOpen();
+            if (current.getPrimaryTerm() < belowTerm) {
+                throw new IllegalArgumentException("Latest translog primary term [" + current.getPrimaryTerm()
+                    + "] has to be not less that trimming term [" + belowTerm + "]");
+            }
+            // update all existed ones (if it is necessary) as checkpoint and reader are immutable
+            final List<TranslogReader> newReaders = new ArrayList<>(readers.size());
+            try {
+                assert current.assertNoSeqAbove(belowTerm, aboveSeqNo);
```

Comment:
can you add a comment as to why we do this? Also, why did you move it under the try catch clause?

Reply:
will add comment; underneath it uses `newSnapshot()` that could lead to `TranslogException` due to `IOException` - but there is no difference with any other exception handling